python/ray/serve/_private/proxy_response_generator.py

```diff
@@ -0,0 +1,147 @@
+import asyncio
+import logging
+import time
+from abc import ABC, abstractmethod
+from asyncio.tasks import FIRST_COMPLETED
+from typing import Any, Callable, Optional, Union
+
+from ray.serve._private.constants import SERVE_LOGGER_NAME
+from ray.serve._private.utils import calculate_remaining_timeout
+from ray.serve.handle import DeploymentResponse, DeploymentResponseGenerator
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class _ProxyResponseGeneratorBase(ABC):
+    def __init__(
+        self,
+        *,
```

Comment:
nit: why do we need to have `*` in this case?

Reply:
this is just style, it forces the following params to be passed as keyword arguments instead of positionals